app/models/framework_response.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class FrameworkResponse < VersionedModel
+  validates :type, presence: true
+
+  belongs_to :framework_question
+  belongs_to :person_escort_record
+  has_many :dependents, class_name: 'FrameworkResponse',
+                        foreign_key: 'parent_id'
+
+  belongs_to :parent, class_name: 'FrameworkResponse', optional: true
+  validates_each :value, on: :update do |record, _attr, _value|
+    if record.parent&.option_selected?(record.framework_question.dependent_value) && record.framework_question.required
+      record.errors.add(:value, :blank)
+    end
+  end
+
+  def self.find_sti_class(type_name)
+    case type_name
+    when 'object'
```

Comment:
This is required because we are defining our own STI classes. Divorcing the class from the type name will be beneficial when exposing those values through the API to clients

Reply:
Rather than overriding this - and also the corresponding `sti_name` method in each descendant class could this let Rails do its default thing and then add a regex or something to clean up the exposed names when it comes to the API part? I'm always suspicious of overriding default Rails behaviour without good reason.

Reply:
Removed custom STI names